### PR TITLE
net-voip/murmur: Fix default config and OpenRC files

### PR DESCRIPTION
**Pull request by DarkDefender**
@0xC0ncord I've done the fixes that I pointed out in https://github.com/gentoo/gentoo/pull/29194

I was going to also update the live ebuild, but I noticed that they have revamped how they do the config files.
There is no `mkini.sh` script anymore and they moved most config things to the `auxiliary_files/` directory.
In addition to this, it seems like the systemd files are not handled by cmake.

So I didn't want to do a update for that without consulting you first.
What are your plans for murmur/mumble >=1.5.0?
On the server side, seems like they have done a lot of cleanup so now there is no trace left of "murmur" anymore and everything seem to be "mumble-server" now. (At least in the config files and server service names)

### CODE REVIEW

**Comment by DarkDefender** [comment]
When this is merged, #29194 can be closed as well I've modified the post install message to be more clear as well.

**Comment by DarkDefender** [comment]
Sounds good to me! :)

Do you want me to take a look at the live ebuild as well or should we just merge this for 1.4.x release?

**Comment by 0xC0ncord** [comment]
> Do you want me to take a look at the live ebuild as well or should we just merge this for 1.4.x release?

If you would like to please go ahead! It will make the 1.5.0 upgrade smoother when it lands. The live ebuild should still be functional although I haven't tested it in a couple months.

Many thanks for your work on this! :)

**Comment by DarkDefender** [comment]
The systemd install path issues seem to be related the build system itself. There is a pull request that fixes it for me:
https://github.com/mumble-voip/mumble/pull/6100

**Comment by DarkDefender** [comment]
@0xC0ncord The build system update for the live ebuild still hasn't been merged yet.
Should I perhaps include that as a patch for the live ebuild or should we just drop the live ebuild update until it is merged?

We could also apply the WIP updates as is as they don't make the ebuild any worse functionality wise.

**Comment by DarkDefender** [comment]
Let me explain the last three commits.

The first one is a fix for a zeroconf mistake in the initd files.

The new one updates the live ebuild to build and work correctly with openrc installs.

The last one is a commit that pulls in the WIP build system changes that will fix the systemd install files for us.
Perhaps we could include these as a patch instead?

**Comment by DarkDefender** [comment]
@0xC0ncord the build system changes needed for the live ebuild has been merged into the mumble master branch.
I updated the live ebuild.

I think this is ready for merging. If you think so as well, I will clean up and squash the commits.

File: net-voip/murmur/murmur-9999.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-9999.ebuild:78]: default*

File: net-voip/murmur/murmur-9999.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-9999.ebuild:81]: manually*

File: net-voip/murmur/murmur-9999.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-9999.ebuild:84]: built-in*

File: net-voip/murmur/murmur-9999.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-9999.ebuild:96]: This needs to be wrapped in `|| die` in case this fails.

File: net-voip/murmur/murmur-9999.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-9999.ebuild:135]: Also need a `|| die` here.

**Review by 0xC0ncord — CHANGES_REQUESTED** [review]
Overall looks good! I haven't tested this just yet, but I found a couple minor typos and missing failsafes.

Thanks for your continued work on this. :)

**Comment by 0xC0ncord** [comment]
I want to reiterate that I plan to have the `murmur` package renamed to `mumble-server` when the next stable version of it is released. That will involve undoing a lot of these stopgap changes where we are renaming back "mumble-server" to "murmur".

**Comment by DarkDefender** [comment]
> I want to reiterate that I plan to have the `murmur` package renamed to `mumble-server` when the next stable version of it is released. That will involve undoing a lot of these stopgap changes where we are renaming back "mumble-server" to "murmur".

Yepp, I know :)

That is why I didn't convert all instances of "mumble-server" to "murmur" as I think that would be more work than it is worth.
It also makes it a lot easier to convert the ebuild to be `mumble-server` with hopefully nothing more than a `s/murmur/mumble-server/g`.

File: net-voip/murmur/murmur-1.4.287-r2.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-1.4.287-r2.ebuild:1]: Please update the copyright year to 1999-2024.

File: net-voip/murmur/murmur-1.4.287-r2.ebuild
Comment (0xC0ncord) [inline net-voip/murmur/murmur-1.4.287-r2.ebuild:27]: For the new 1.4.287-r2 version, change `KEYWORDS` to `"~amd64 ~arm ~arm64 ~x86"` so we're not committing this new version straight to stable (it needs time in testing first).

**Review by 0xC0ncord — CHANGES_REQUESTED** [review]
Looks good after testing! Just some last minute cleanup and I think it's ready.

After addressing the review comments please squash your commits into two different commits, one which fixes up the live ebuild and one which adds version `1.4.287-r2`. Please make sure to add the package prefix to the commit summary, e.g.:

```
net-voip/murmur: update live ebuild
net-voip/murmur: add 1.4.287-r2
```

You can leave the old commit summaries in the commits' long descriptions so we know what was changed. ;)

Thanks!

**Comment by DarkDefender** [comment]
I noticed that I needed to add `3rdparty/cmake-compiler-flags` to `EGIT_SUBMODULES` to get both the live build of murmur and mumble to build.

Is this something you want to take care of or investigate yourself?

**Comment by 0xC0ncord** [comment]
> I noticed that I needed to add `3rdparty/cmake-compiler-flags` to `EGIT_SUBMODULES` to get both the live ebuild of murmur and mumble to build.
> 
> Is this something you want to take care of or investigate yourself?

I just took a quick look at it and it seems fine -- it's not a library or anything, just some CMake utility functions I assume the build system is now using.

**Comment by DarkDefender** [comment]
I think this should be ready for merging now.

**Comment by DarkDefender** [comment]
@0xC0ncord anything holding this up?

**Comment by DarkDefender** [comment]
Fixed the missing unstable keyword :)

**Comment by 0xC0ncord** [comment]
All done. Thanks for your patience and dedication to getting fixing this up. :)

**Pull request closed without merging**
